src/socket_poller.cpp

```diff
@@ -67,15 +78,10 @@ zmq::socket_poller_t::~socket_poller_t ()
     tag = 0xdeadbeef;
 
     for (items_t::iterator it = items.begin (); it != items.end (); ++it) {
-        if (it->socket && it->socket->check_tag ()) {
-            int thread_safe;
-            size_t thread_safe_size = sizeof (int);
-
-            if (it->socket->getsockopt (ZMQ_THREAD_SAFE, &thread_safe,
-                                        &thread_safe_size)
-                  == 0
-                && thread_safe)
-                it->socket->remove_signaler (signaler);
+        // TODO shouldn't this zmq_assert (it->socket->check_tag ()) instead?
```

Comment:
could you please open an issue? the TODO is just going to get forgotten I'm afraid